### PR TITLE
 mirror images via oc-mirror and Upgrade catalog qe-app-registry after ocp upgrade

### PR DESCRIPTION
**Pull request by anpingli**
Recall enable-qe-catalog step to upgrade aosqe-index after upgrade.
Only mirror operator header to reduce the image size(now the size is less than 20 Gi on C2S cluster) 
Warning about testing:
1) Should to pass the test on the supported minimal version of OCP  on QE jobs
2) should to pass the test on one of the upgrade
3) should to pass the test on one of  the C2S job

### CODE REVIEW

File: ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh [hunk 1]
```diff
@@ -478,6 +478,36 @@ function check_upgrade_status() {
     fi
 }
 
+# update qe-app-registry catalogsource to correct version after OCP is upgraded
+function update_aosqe_index() {
+    if oc get catalogsource qe-app-registry -n openshift-marketplace >/dev/null 2&1 ; then
+        ocp_channel=$(oc get clusterversion/version -o jsonpath="{.spec.channel}")
+        if [[ $? == 0 ]] ; then
+            ocp_version=${ocp_channel//stable-}
```
Comment (jianzhangbjz) [inline ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:486]: This channel value is not `stable-` prefix sometimes, so I guess we can get the version by using
```console
oc version | grep "Server Version" |awk '{print $3}'
```

Comment (anpingli) [reply ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:486]: Acked

File: ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh [hunk 2]
```diff
@@ -223,7 +224,7 @@ function set_CA_for_nodes () {
 
 # Create the fixed ICSP for optional operators
 function create_settled_icsp () {
-    cat <<EOF | oc create -f -
+    cat <<EOF | oc apply --validate='ignore' -f -
```
Comment (jianzhangbjz) [inline ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh:227]: Why skip the schema validation here?

Comment (anpingli) [reply ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh:227]: This flag can disable the warning message below:
Warning: resource xxx/xxx is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by oc apply. oc apply should only be used on resources created declaratively by either oc create --save-config or oc apply. The missing annotation will be patched automatically.

**Comment by anpingli** [comment]
@jianzhangbjz  @jiajliu   @yunjiang29  a new mirror step is called in disconnect cluster. PTAL

File: ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml [hunk 3]
```diff
@@ -8,5 +8,6 @@ chain:
   - ref: cucushift-upgrade-toimage
   - ref: cucushift-upgrade-check
   - ref: cucushift-upgrade-report
+  - ref: enable-qe-catalogsource
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml:11]: We need do post-upgrade test based on the updated qe-app-registry, right? if so, should place it before cucushift-upgrade-check

File: ci-operator/step-registry/cucushift/upgrade/disconnected/cucushift-upgrade-disconnected-chain.yaml [hunk 4]
```diff
@@ -9,5 +9,6 @@ chain:
   - ref: cucushift-upgrade-toimage
   - ref: cucushift-upgrade-check
   - ref: cucushift-upgrade-report
+  - ref: enable-qe-catalogsource-disconnected
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/upgrade/disconnected/cucushift-upgrade-disconnected-chain.yaml:12]: the same ^

File: ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml (hunk 3 above)
Comment (jiajliu) [reply ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml:11]: Following chains are also have post-upgrade test, so we needed the step too?
```
- chain: openshift-upgrade-qe-test
- chain: openshift-upgrade-qe-test-cert-rotation
- chain: openshift-upgrade-qe-test-heterogeneous
- chain: openshift-upgrade-qe-test-disconnected
- chain: openshift-upgrade-qe-test-paused-to-latest
- chain: openshift-upgrade-qe-test-paused
```

Comment (anpingli) [reply ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml]: Done

File: ci-operator/step-registry/cucushift/upgrade/disconnected/cucushift-upgrade-disconnected-chain.yaml (hunk 4 above)
Comment (anpingli) [reply ci-operator/step-registry/cucushift/upgrade/disconnected/cucushift-upgrade-disconnected-chain.yaml]: Done

File: ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml (hunk 3 above)
Comment (jiajliu) [reply ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml]: @anpingli I only see `chain: openshift-upgrade-qe-test` updated, but not for others, or you mean no need for other upgrade chains, double check?
```
- chain: openshift-upgrade-qe-test
- chain: openshift-upgrade-qe-test-cert-rotation
- chain: openshift-upgrade-qe-test-heterogeneous
- chain: openshift-upgrade-qe-test-disconnected
- chain: openshift-upgrade-qe-test-paused-to-latest
- chain: openshift-upgrade-qe-test-paused
- chain: openshift-upgrade-qe-test-disconnected-cert-rotation
```

**Comment by jiajliu** [comment]
cc ota guys for the new universal step.
/cc @jianlinliu @shellyyang1989 @JianLi-RH

File: ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml (hunk 3 above)
Comment (anpingli) [reply ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml]: It seems i didn't submit the changes.

Comment (anpingli) [reply ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml]: Done. Please refresh the page.

File: ci-operator/step-registry/openshift/upgrade/qe/test/disconnected/openshift-upgrade-qe-test-disconnected-chain.yaml [hunk 5]
```diff
@@ -9,6 +9,7 @@ chain:
   - ref: cucushift-upgrade-mirror-images-by-oc-adm
   - ref: cucushift-upgrade-healthcheck
   - ref: cucushift-upgrade-toimage
+  - ref: enable-qe-catalogsource
```
Comment (anpingli) [inline ci-operator/step-registry/openshift/upgrade/qe/test/disconnected/openshift-upgrade-qe-test-disconnected-chain.yaml:12]: @yunjiang29  Shall we call enable-disconnect-qe-catalogsource in test-disconnected-chain? How we distinguish C2S clusters?

File: ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh (hunk 2 above)
Comment (jianzhangbjz) [reply ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh:227]: So, will it block the installation if use the `validate`? I don't think skip the validation is a good solution.

File: ci-operator/step-registry/openshift/upgrade/qe/test/disconnected/openshift-upgrade-qe-test-disconnected-chain.yaml (hunk 5 above)
Comment (jianzhangbjz) [reply ci-operator/step-registry/openshift/upgrade/qe/test/disconnected/openshift-upgrade-qe-test-disconnected-chain.yaml:12]: I guess so. The `enable-qe-catalogsource-disconnected` step will check if the cluster is the `C2S` one, here: https://github.com/openshift/release/blob/master/ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh#L380

Comment (yunjiang29) [reply ci-operator/step-registry/openshift/upgrade/qe/test/disconnected/openshift-upgrade-qe-test-disconnected-chain.yaml:12]: > I guess so. The `enable-qe-catalogsource-disconnected` step will check if the cluster is the `C2S` one, here: https://github.com/openshift/release/blob/master/ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh#L380

Right.

Comment (yunjiang29) [reply ci-operator/step-registry/openshift/upgrade/qe/test/disconnected/openshift-upgrade-qe-test-disconnected-chain.yaml:12]: `enable-qe-catalogsource-disconnected` should be used

**Comment by anpingli** [comment]
@zhouying7780 PTAL, @jiajliu @jianzhangbjz  PTAL again

**Comment by anpingli** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-stable-4.12-aws-usgov-ipi-private-f28 periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-c2s-ipi-disconnected-private-fips-f14

**Comment by anpingli** [comment]
/pj-rehearse  periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-aws-c2s-ipi-disconnected-private-fips-f28

**Comment by anpingli** [comment]
Enable qe-catalogsource pass on 
ci/rehearse/openshift/azure-disk-csi-driver/master/e2e-azure-csi-extended and ci/rehearse/openshift/azure-disk-csi-driver/release-4.15/e2e-azure-csi-extended

**Comment by anpingli** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-aws-c2s-ipi-disconnected-private-fips-f28 periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-vsphere-ipi-sdn-p2-f28

**Comment by anpingli** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-aws-c2s-ipi-disconnected-private-fips-f28 periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-vsphere-ipi-sdn-p2-f28

**Comment by anpingli** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-stable-aws-c2s-ipi-disconnected-private-fips-f28

**Comment by anpingli** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-vsphere-ipi-sdn-p2-f28

**Comment by anpingli** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-stable-aws-c2s-ipi-disconnected-private-fips-f28

**Comment by anpingli** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-azure-upi-f28

**Comment by anpingli** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-azure-upi-f28

**Comment by anpingli** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-aws-c2s-ipi-disconnected-private-fips-f28

**Comment by anpingli** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-xpn-f28-

**Comment by anpingli** [comment]
Close to the PR. as I want to make the PR simple. Split into two parts. one for installer another for ugprade.

**Pull request closed without merging**